spec/mail/messagedelivery.mdown

```diff
@@ -0,0 +1,221 @@
+
+If an object has been modified AND deleted since the oldState, the server SHOULD just return the id in the *removed* response, but MAY return it in the changed response as well. If an object has been created AND deleted since the oldState, the server SHOULD remove the object id from the response entirely, but MAY include it in the *removed* response, and (if in the *removed* response) MAY included it in the *changed* response as well.
+
+The following errors may be returned instead of the *messageDeliveryUpdates* response:
+
+`accountNotFound`: Returned if an *accountId* was explicitly included with the request, but it does not correspond to a valid account.
+
+`accountNotSupportedByMethod`: Returned if the *accountId* given corresponds to a valid account, but the account does not support this data type.
+
+`invalidArguments`: Returned if the request does not include one of the required arguments, or one of the arguments is of the wrong type, or otherwise invalid. A *description* property MAY be present on the response object to help debug with an explanation of what the problem was.
+
+`cannotCalculateChanges`: Returned if the server cannot calculate the changes from the state string given by the client. Usually due to the client's state being too old, or the server being unable to produce an update to an intermediate state when there are too many updates. The client MUST invalidate its cache for this object type.
+
+## setMessageDeliveries
+
+This method is used to send messages, or to recall a sent message (if possible). It takes the following arguments
+
+- **accountId**: `String|null`
+  The id of the account to use for this call. If not given, defaults to the primary account.
+- **ifInState**: `String|null`
+  This is a state string as returned by the *getMessageDeliveries* method. If supplied, the string must match the current state, otherwise the method will be aborted and a `stateMismatch` error returned. If `null`, any changes will be applied to the current state.
+- **create**: `String[Message]|null`
+  A map of *creation id* (an arbitrary string set by the client) to MessageDelivery objects to create. Note, the *messageId* may be a back-reference to a message created in the same API request.
+
+    This is how a message is sent.
+
+- **destroy**: `String[]|null`
+  A list of ids for MessageDelivery objects to delete.
+
+    This is how a message may be unsent.
+
+- **onCreateUpdateMessage**: `String[Message]`
+  A map of *creation id* to an object containing properties to update on the Message object referenced by the MessageDelivery if the create succeeds.
+- **onDestroyUpdateMessage**: `String[Message]`
+  A map of *MessageDelivery id* to an object containing properties to update on the Message object referenced by the MessageDelivery if the destroy succeeds.
```

Comment:
Updating a message is cool.

We should be also able to delete them. IE: delete draft upon send.